src/Emulator/Cores/Arm-M/CortexM.cs

```diff
@@ -53,6 +54,36 @@ public override void Resume()
 
         public override string Architecture { get { return "arm-m"; } }
 
+        public override List<IGBDFeature> GDBFeatures
+        {
+            get
+            {
+                List<IGBDFeature> features = new List<IGBDFeature>();
+
+                IGBDFeature feature = new IGBDFeature("org.gnu.gdb.arm.m-profile");
+                for(uint index = 0; index <= 12; index++)
+                {
+                    feature.Registers.Add(new IGBDRegister(index, 32, $"r{index}", "uint32", "general"));
+                }
+                feature.Registers.Add(new IGBDRegister(13, 32, "sp", "data_ptr", "general"));
+                feature.Registers.Add(new IGBDRegister(14, 32, "lr", "uint32", "general"));
+                feature.Registers.Add(new IGBDRegister(15, 32, "pc", "code_ptr", "general"));
+                feature.Registers.Add(new IGBDRegister(25, 32, "xpsr", "uint32", "general"));
+                features.Add(feature);
+
+                feature = new IGBDFeature("org.gnu.gdb.arm.m-system");
```

Comment:
For clarity I suggest avoid reusing the variable. Instead create a new one with a distinct name, like:
```suggestion
                var mSystemFeature = new IGBDFeature("org.gnu.gdb.arm.m-system");
```